tests/rptest/services/admin.py

```diff
@@ -157,12 +157,14 @@ def _get_stable_configuration(
                     f"get status:{meta['status']} while already observed:{status} before"
                 )
                 return None
-            read_replicas = set([r['node_id'] for r in meta["replicas"]])
+            read_replicas = meta["replicas"]
             if replicas is None:
                 replicas = read_replicas
                 self.redpanda.logger.debug(
                     f"get replicas:{read_replicas} from {host}")
-            elif replicas != read_replicas:
+            elif {(r["node_id"], r["core"])
```

Comment:
it's too complex for a condition, do something e.g. extract it into a nested function